tools/targets/__init__.py

```diff
@@ -379,6 +379,28 @@ def labels(self):
         labels = (names + CORE_LABELS[self.core] + self.extra_labels)
         return labels
 
+    @property
+    def core_without_NS(self):
+        if self.core.endswith('-NS'):
+            return self.core[:-3]
+        else:
+            return self.core
+
+    # Mechanism for specifying TrustZone is subject to change - see
+    # discussion on https://github.com/ARMmbed/mbed-os/issues/9460
+    # In the interim, we follow heuristics that support existing
+    # documentation for ARMv8-M TF-M integration (check the "TFM" label),
+    # plus an extra "trustzone" flag set by M2351, and looking at the "-NS"
+    # suffix. This now permits non-TrustZone ARMv8 builds if
+    # having trustzone = false (default), no TFM flag, and no -NS suffix.
+    @property
+    def is_TrustZone_secure_target(self):
+        return (self.trustzone or 'TFM' in self.labels) and not self.core.endswith('-NS')
```

Comment:
You could add
     "trustzone": true,
in SPE_Target, and remove this TFM check here ?

Reply:
Sadly no, because some SPE_Targets are not using TrustZone. See CY8CKIT_062_WIFI_BT_M0_PSA.

Dual-CPU set-up with the secure and non-secure images running on different cores with message-passing.

I could add `trustzone` to all the ARMv8-M SPE_Targets, but that would need to be added to the documentation of creating a TF-M target, and I'm wary about breaking some out-of-tree target I haven't seen. This ensures that the existing instructions work.

And I'm not going to make it "SPE_Target and core arch >= 8", because there's no reason you can't run the same dual-cpu PSA setup with an M0 and M33 - if those guys want a slight CPU upgrade from M4 without a complete system rearchitecture.

Reply:
I guess I'm also wary of documenting the trustzone flag as part of the TF-M instructions, if we think it might be subject to change. It's part of this being the "quick" fix - avoiding discussions about what the targets.json should look like.

If we're willing to commit to it, I could add it, so that the "detect TFM label" is then clearly legacy support - TFM targets are supposed to also have the trustzone flag set properly.

Reply:
If you don't add trustzone to true in SPE_Target,
maybe we got a strange situation where trustzone value is kept to the default value, i.e. false?

Reply:
Yes, and that's strange if it is a TFM build. (Not if it isn't).

But the trustzone value is not directly exposed anywhere. The `Target.is_TrustZone_xxx` properties will give the right answer - looking at both that flag and the `TFM` label.

I can't add the flag to `SPE_Target` because not all SPE Targets are trustzone.

Although maybe I could and set it back to false in the exception cases, as it's TFM which is the "normal" case? So PSOC6 inherits from SPE_Target and flips trustzone back to false?

(I'm kind of playing games here trying to avoid changing any documentation or existing publicly visible behaviour so this can slip into 5.12.y as a "target update".)